java/listenup-solution-server/src/main/java/com/twilio/interview/listenup/UserService.java

```diff
@@ -0,0 +1,133 @@
+package com.twilio.interview.listenup;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.gson.Gson;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.StatusLine;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+public class UserService {
+
+  private final String friendsURLPrefix = "http://localhost:8000";
+  private final String playsURLPrefix = "http://localhost:8001";
+  private Map<String, User> users = new HashMap<>();
+
+  private String getJsonResponse(String url) throws ClientProtocolException, IOException {
+    CloseableHttpClient httpclient = HttpClients.createDefault();
+    HttpGet httpGet = new HttpGet(url);
+    CloseableHttpResponse response1 = httpclient.execute(httpGet);
+    StatusLine sl = response1.getStatusLine();
+    int statusCode = sl.getStatusCode();
+    if (statusCode == 404) {
+      return null;
+    }
+    try {
+      HttpEntity entity1 = response1.getEntity();
+      String json = EntityUtils.toString(entity1);
+      return json;
+    } finally {
+      response1.close();
+    }
+  }
+
+  private ServiceResponse getServiceRequest(String url) throws ClientProtocolException, IOException {
+    String json = getJsonResponse(url);
+    Gson gson = new Gson();
```

Comment:
If you're constantly instantiating `Gson` then just create it as a static variable at the top 
`private static final Gson GSON = new Gson();`